jgrapht-core/src/test/java/org/jgrapht/perf/matching/PathGrowingWeightedMatchingPerformanceTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.perf.matching;
+
+import java.util.concurrent.TimeUnit;
+
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.VertexFactory;
+import org.jgrapht.alg.EdmondsBlossomShrinking;
+import org.jgrapht.alg.interfaces.MatchingAlgorithm;
+import org.jgrapht.alg.matching.GreedyWeightedMatching;
+import org.jgrapht.alg.matching.PathGrowingWeightedMatching;
+import org.jgrapht.generate.GnpRandomGraphGenerator;
+import org.jgrapht.generate.GraphGenerator;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.Pseudograph;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import junit.framework.TestCase;
+
+/**
+ * A small benchmark comparing matching algorithms.
+ * 
```

Comment:
I think (1) is a bit difficult without explicitly copying the class, or using some external script.

(2) is what the performance tests are doing, but still jmh only measures time. I initially wanted to add a test which compares the quality of the matchings (compared to optimal) but found that jmh is not designed for that. They have something called `AuxCounters` but they are somehow not the right way to go. 

As a first step, I think we should separate the benchmarks in a separate package called jgrapht-perf.